docs/DafnyRef/Attributes.md

```diff
@@ -623,6 +623,11 @@ method doSplitHere(x: bool) returns (y: int) {
 ### 23.3.3. `{:subsumption n}`
 Overrides the `/subsumption` command-line setting for this assertion.
 
+### 23.3.4. `{:expect s}`
```

Comment:
For the future, it seems like we could also support adding this to other specification contexts like `requires`, `ensures`, `invariant`, etc. (although see my independent question about whether this is the name we want)

Reply:
Despite my slack comment, I see your point. I'm open to any naming, but making it anything other than an attribute would be more work and harder to generalize.

Reply:
On thinking about it more I like using a `{:checked}` attribute. I think that terminology fits in well with the documentation for the related `--test-assumptions` option (in fact I even wonder if the option should be `check-assumptions` to avoid implying it's strongly coupled to `dafny test`. 

We could think of `expect P;` as a short form for `assume {:checked} P;` (that's actually allowed when compiling).

Interested in @atomb's option here too.

Reply:
What about {:check} to match assert, assume, expect as active present tense verbs.

Reply:
Another thought: perhaps it would be better if `expect` itself behaved this way. That is, if it had to be both verified and checked at runtime. We could change this for Dafny 4.0.

I submit that even when writing a `{:test}` in Dafny, you almost always believe the expression you're `expect`-ing to be true. Tests often look a lot like examples, and examples in Dafny should verify as well, i.e. if you can't satisfy the precondition of the function you're trying to call, it's often an indication the specified interface of the code you're testing is wrong or too week.

We would also allow `expect {:axiom} P;` just like `assert {:axiom} P;`, for easy backwards compatibility and those times it's too much work to prove `P`. `:- expect {:axiom} ...` as well.

Reply:
@seebees I'm curious whether what I'm saying above jives with your more substantial experience than mine in using `expect` in Dafny tests.

Reply:
For the set of contracts currently supported, `--check-assumptions` would be a nice improvement on naming. However, I worry it might be confusing if we extend it to evaluate other contracts at runtime, including those that, at least in principle, might be verified. What do others think?

Reply:
Note there is deeper discussion happening on the issue now, so we should move further replies there: https://github.com/dafny-lang/dafny/issues/3410#issuecomment-1419618347